sdk/python/feast/data_source.py

```diff
@@ -471,33 +479,105 @@ def get_table_column_names_and_types(
     ) -> Iterable[Tuple[str, str]]:
         pass
 
+    def __eq__(self, other):
+        if not isinstance(other, RequestSource):
+            raise TypeError(
+                "Comparisons should only involve RequestSource class objects."
+            )
+        if (
+            self.name != other.name
+            or self.description != other.description
+            or self.owner != other.owner
+            or self.tags != other.tags
+        ):
+            return False
+        else:
+            if isinstance(self.schema, List) and isinstance(other.schema, List):
+                for field1, field2 in zip(self.schema, other.schema):
+                    if field1 != field2:
+                        return False
+                return True
+            elif isinstance(self.schema, Dict) and isinstance(other.schema, Dict):
```

Comment:
You should check here that `set(self.schema.keys()).symmetric_difference(set(other.schema.keys())` is empty; otherwise this check will incorrectly pass if `other.schema` has keys that `self.schema` does not.